cutils.h

```diff
@@ -356,6 +357,81 @@ static inline void inplace_bswap32(uint8_t *tab) {
     put_u32(tab, bswap32(get_u32(tab)));
 }
 
+static inline double fromfp16(uint16_t v) {
```

Comment:
Where does one find the rules you followed here? (trying to inform myself and do a good review 😅 )

Reply:
I mostly derived it from first principles; they're identical to floats and doubles, except with smaller exponents and mantissas (5 and 10 bits resp. with an exponent bias of 15. Compare doubles: 11 and 52 bits, bias of 1023.)

To get the floating point value, you compute `pow(2, exponent-15) * (1 + mantissa/1024.0)`. Compare doubles: `pow(2, exponent-1023) * (1 + mantissa/pow(2, 52))`

To go from double to half-float or vice versa... my intuition was that `exponent - 1023 + 15` and `mantissa >> (52 - 10)` would be Good Enough(TM) but alas, it wasn't, so now I'm mostly computing it in the double-precision domain.

The [Wikipedia page](https://en.wikipedia.org/wiki/Half-precision_floating-point_format) is okay.

Reply:
Oh, and if you're worried about correctness: test262 has pretty good coverage so it should be okay. Everything passes.

Reply:
Thank you!